src/CMakeLists.txt

```diff
@@ -39,3 +39,11 @@ set (src_DIST
    ${src_kms_message_DIST}
    PARENT_SCOPE
 )
+
+
+if (CMAKE_CXX_COMPILER)
+   # Add a C++ source file that will #include the main C headers. This "library"
+   # does nothing other than validate that the C headers are valid C++ headers.
+   add_library (mongoc-cxx-check STATIC cpp-check.cpp)
+   target_link_libraries (mongoc-cxx-check PRIVATE mongoc_static)
+endif()
```

Comment:
You will also need this change to fix the failure in the `make-release-archive` task:

```
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1876b2224..5a49fd8ed 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -29,6 +29,7 @@ file (GLOB_RECURSE src_kms_message_DIST RELATIVE "${SOURCE_DIR}" kms-message/*)
 
 set_local_dist (src_DIST_local
    CMakeLists.txt
+   cpp-check.cpp
 )
 
 set (src_DIST
```